tools/agent_tool/src/cpp/main.cpp

```diff
@@ -0,0 +1,224 @@
+
+            case optionIndex::TRANSPORT:
+                if (strcmp(opt.arg, "tcp") == 0)
+                {
+                    transport_protocol = eprosima::ddsrouter::core::types::TransportProtocol::tcp;
+                }
+                else if (strcmp(opt.arg, "udp") == 0)
+                {
+                    transport_protocol = eprosima::ddsrouter::core::types::TransportProtocol::udp;
+                }
+                break;
+
+            case optionIndex::UNKNOWN_OPT:
+                std::cerr << "ERROR: " << opt.name << " is not a valid argument." << std::endl;
+                option::printUsage(fwrite, stdout, usage, columns);
+                return 1;
+                break;
+        }
+    }
+
+    {
+        if (entity_type == CLIENT)
+        {
+            domain = connection_address;
+            logUser(AMLIPCPP_MANUAL_TEST, "Starting Manual Agent Node execution. Creating connection address...");
+        }
+        else
+        {
+            domain = listening_address;
+            logUser(AMLIPCPP_MANUAL_TEST, "Starting Manual Agent Node execution. Creating listening address...");
+        }
+
+        // Create connection address
+        auto address = eprosima::ddsrouter::core::types::Address(
+            12121,
+            12121,
+            domain,
+            transport_protocol);
+
+        try
+        {
+            switch (entity_type)
+            {
+                case CLIENT:
+                {
+                    logUser(AMLIPCPP_MANUAL_TEST, "Address to connect: " << address << ". Creating Node...");
+
+                    // Create Client Node
+                    eprosima::amlip::node::agent::ClientNode client_node(
+                        name,
+                        { address });
+                    break;
+                }
+                case SERVER:
+                {
+                    logUser(AMLIPCPP_MANUAL_TEST, "Address where listen: " << address << ". Creating Node...");
+
+                    // Create Server Node
+                    eprosima::amlip::node::agent::ServerNode server_node(
+                        name,
+                        { address });
+                    break;
+                }
+                case REPEATER:
+                {
+                    logUser(AMLIPCPP_MANUAL_TEST, "Address where listen: " << address << ". Creating Node...");
+
+                    // Create Turn Node
+                    eprosima::amlip::node::agent::TurnNode turn_node(
+                        name,
+                        { address });
+                    break;
+                }
+            }
+        }
+        catch(const std::exception& e)
+        {
+            std::cerr << "Execution failed with error:\n " << e.what() << std::endl;
+        }
+
+        logUser(AMLIPCPP_MANUAL_TEST, "Node created. Waiting SIGINT (C^) to close...");
+
+        eprosima::utils::event::SignalEventHandler<eprosima::utils::event::Signal::sigint> sigint_handler;
```

Comment:
Maybe create handler at the beggining, so it can be closed before everything is built